File: queue/queue.js
class Queue {
    constructor(){
        this.queue = []
    }
    enqueue(data){
        this.queue.push(data)
    }
    dequeue(){
        if (this.isEmpty()) {
            return "Queue is Empty"
        }
        this.queue.shift()
    }
    isEmpty(){
       return this.queue.length === 0
    }
    peek(){
        if (this.isEmpty()) {
            return "Queue is Empty"
        }
        return this.queue[0]
    }
    size(){
        this.queue.length()
    }
    clear(){
        this.queue=[]
    }
    contain(data){
        this.queue.includes(data)
    }
    printQueue(){
        let str=""
        for(let i =0;i<=this.queue.length;i++){
            str += this.queue[i] + "\n"
        }
        return str
    }
}

let myq = new Queue
myq.enqueue(3)
myq.enqueue(242)
myq.enqueue(534)
myq.enqueue(343)
 
console.log(myq.printQueue());
myq.dequeue()

console.log(myq.printQueue());

File: queue/queue.py
class Queue:
    def __init__(self):
        self.queue=[]
    
    def enqueue(self,data):
         return self.queue.append(data)  # can also use insert instead of appnd

    def dequeue(self):
        if(self.queue == []):
            print("Queue is Empty")
        return self.queue.pop()
    
    def isEmpty(self):
       return self.queue == [] 
    
    def peek(self):
        return self.queue[0]
    
    def size(self):
        return self.queue.__len__
    
    def contain(self,key):
        return key in self.queue
    def PRintft(self):
        return str(self.queue)

q = Queue()
q.enqueue(2)
q.enqueue(42)
q.enqueue(33)
q.enqueue(34)

print(q.isEmpty())

print(q.contain(42))
print(q.PRintft())